preload/resources/fetch-destination-worker.js

```diff
@@ -0,0 +1,11 @@
+self.addEventListener('fetch', function(event) {
+    if (event.request.url.indexOf('dummy.xml') != -1) {
+        var response = new Response();
+        if (event.request.destination == "")
+            response.status = 200;
+        else
+            response.status = 500;
```

Comment:
In theory returning a network error can be done through `Response.error()` per https://fetch.spec.whatwg.org/#dom-response-error.